### ISSUE TITLE
[Bug]: HTTP Information Not Available In X Ray Headers

### ISSUE THREAD

**Issue by 5t33** [aws-observability/aws-rum-web#427]
### Which web client version did you detect this bug with?

v1.14.0

### What environment (build systems, module system, and framework) did you detect this bug with?

TypeScript v4.7.4, node v19.7.0, React ^17.0.0

### Is your web application a single page application (SPA) or multi page application (MPA)?

None

### Please provide your web client configuration

```
      const config: AwsRumConfig = {
        allowCookies: true,
        enableXRay: true,
        guestRoleArn: awsRumGuestRoleArn,
        identityPoolId: awsRumIdentityPoolId,
        sessionSampleRate: sessionSampleRate,
        telemetries: [
          'errors', 
          [ "performance", {recordAllTypes: ['document', 'script', 'other']} ],
          [ 'http', {
            recordAllRequests: true,
            addXRayTraceIdHeader: true,
            urlsToExclude: [
              new RegExp('/audio-samples/.+'),
              new RegExp('https://dataplane.rum.us-west-2.amazonaws.com/.+')
            ]
          } ]],
        eventPluginsToLoad: []
      };
```

### Please describe the bug/issue

Hi, 

I'm seeing HTTP traces coming from my app in my X Ray console, however, they lack any headers that allow me to sort or filter them.  

I have X Ray set up on the backend with this package: https://github.com/narando/nest-xray. For reference, this is how my X Ray console looks without the RUM X Ray segments: 

<img width="1406" alt="Screenshot 2023-07-15 at 5 53 30 PM" src="https://github.com/aws-observability/aws-rum-web/assets/19535720/9b9f1bbc-f786-40ef-a480-d0b34591f107">

As you can see, URLs and status codes are sortable & filterable. However, when I implement RUM, I get this:
<img width="1419" alt="Screenshot 2023-07-15 at 6 02 51 PM" src="https://github.com/aws-observability/aws-rum-web/assets/19535720/5964c7a3-435d-422f-ad2a-0169ac8c0874">

Only trace IDs. And when you dig into the trace, you can see that the top level of the trace doesn't include the HTTP info:
<img width="1141" alt="Screenshot 2023-07-14 at 2 17 55 PM" src="https://github.com/aws-observability/aws-rum-web/assets/19535720/59c32fa0-4b98-420c-b42d-967760519bae">

It's somehow obscured by the parent "app monitor" trace. 

Is this something in my set up?

**Comment by haotiantest** [comment]
Hey Steven, in the trace tables, they are showing segment infomations, but when RUM web client generate traces, we put http object in the sub-segment document, that's why it's not showing in the table but if you look at the raw segment data you will able to see it there. We will evluate to see if we can add http object into segment level as well.

**Comment by qhanam** [comment]
I agree with @5t33 -- I think this is worth fixing. The top-level trace segment is effectively the HTTP request, and so it should contain all the properties of the HTTP request.

As shown in the screenshot, the X-Ray console only shows the status/response time/http method/url of the root trace segment. Among other things, this makes it difficult to select a trace for debugging purposes.

**Comment by flochaz** [comment]
Inspecting the raw trace, I am wondering if it is not linked to the fact that root segment is not in it. So the issue is probably due to the fact that The root segment that should come from RUM is not received by XRay. (PS: I am in eu-west-3 region ... might as well be a source of issue for rum -> xray communication ?)

**Comment by flochaz** [comment]
On my side the issue was the way I was setting my regex for `addXRayTraceIdHeader`. Not sure what was wrong with `new RegExp` constructor but going straight for a regex like `	addXRayTraceIdHeader: [/api\.\w+\.example\.com/],` worked
